On newer kernels there are more optional parameter fields
before the hyphen

---
 libinit.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- libinit.c
+++ libinit.c	2019-04-09 12:47:35.913748799 +0000
@@ -281,7 +281,7 @@ static void init_mounts(void)
 
     if ((mnt = fopen("/proc/self/mountinfo", "re")) == (FILE*)0)
 	return;
-    while (fscanf(mnt, "%i %i %u:%u %*s %s %*s - %s %*s %*[^\n]", &mid, &parid, &maj, &min, &point[0], &fstype[0]) == 6) {
+    while (fscanf(mnt, "%i %i %u:%u %*s %s %*[^-] - %s %*s %*[^\n]", &mid, &parid, &maj, &min, &point[0], &fstype[0]) == 6) {
 	const size_t nlen = strlen(point);
 	MNTINFO *restrict p;
 	if (posix_memalign((void*)&p, sizeof(void*), alignof(MNTINFO)+(nlen+1)) != 0) {
